app/services/usage_monitoring/process_organization_subscription_activities_service.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module UsageMonitoring
+  class ProcessOrganizationSubscriptionActivitiesService < BaseService
+    Result = BaseResult[:nb_jobs_enqueued]
+
+    BATCH_SIZE = 500
+
+    def initialize(organization:)
+      @organization = organization
+      super()
+    end
+
+    def call
+      nb_jobs_enqueued = 0
+
+      # NOTE: If we need to handle different delays per organization, this would be done here.
+
+      organization.subscription_activities.where(enqueued: false).select(:id).in_batches(of: BATCH_SIZE) do |batch|
+        jobs = []
+        batch.each do |subscription_activity|
+          jobs << ProcessSubscriptionActivityJob.new(subscription_activity.id)
+        end
+
+        ActiveRecord::Base.transaction do
+          batch.update_all(enqueued: true, enqueued_at: Time.current) # rubocop:disable Rails/SkipsModelValidations
+          after_commit { ActiveJob.perform_all_later(jobs) }
```

Comment:
`perform_all_later` does bypass the AJ-uniqueness I think, (it's not configured for this job, but perhaps you could end up enqueuing lots of jobs here)

Reply:
Yeah, I didn't use activejob-uniqueness on purpose. With the `enqueued` flag, I don't expect duplicate jobs and if there is, the `subscription_activity_id` won't be found so the job will just exit.

Reply:
Also one more potential problem is that you can have a job enqueued that dies (for some reason) and so you can end up with an `enqueued` record preventing any further processing for that subscription. Probably wise to setup a safeguard for that case as well.